src/unix/core.c

```diff
@@ -1444,3 +1444,17 @@ int uv_gettimeofday(uv_timeval_t* tv) {
   tv->tv_usec = time.tv_usec;
   return 0;
 }
+
+int64_t uv_get_usec_since_epoch() {
+  int64_t ret;
+  struct timeval time;
+
+  if (gettimeofday(&time, NULL) != 0)
+    return UV__ERR(errno);
+
+	/* cast to unsigned to prefer Y2K38 over dates < 1970 */
+	ret = (uint64_t)time.tv_sec * 1000000;
```

Comment:
`int64_t`?

Reply:
on "03:14:08 UTC on 19 January 2038" time.tv_sec will equal `b‭1000'0000'0000'0000'0000'0000'0000'0000‬` so we can interpret that either as and unsigned `2147483648` or a signed `-2147483648`.

Reply:
https://coliru.stacked-crooked.com/a/1ad3777fd9b0197a

Reply:
Yes, casting was wrong.